### DISCUSSION TITLE
How to add a python library in odoo/custom/dependencies/pip.txt and compile again?

### DISCUSSION THREAD

**Discussion by pleirb** [Tecnativa/doodba-copier-template#399] [Q&A] [answered]
I'm trying to add a python library hosted on Github like this:
`git+https://github.com/MyProyect/moduleX.git@main`
I save and run **invoke img-build** but can't get it to install. What am I doing wrong?

**Comment by pedrobaeza** [comment]
Is it a standard Python library? Which is the error?

**Reply by pleirb** [reply]
It is a custom library and it seems to be installed correctly (I do a pip show pyXXX, the library data comes out fine) but in the odoo server log, through a module that is also custom, it shows: ModuleNotFoundError: No module named 'pyXXX '

**Reply by pedrobaeza** [reply]
Maybe you are not using the proper rebuilt image? It should work according the steps you are providing.

**Reply by pleirb** [reply]
Yeah, I don't know what's up. Normally, the library is installed by doing pip install -r requirements.txt where the content of requirements.txt is as follows:
git+https://github.com/MyProject/moduleX.git@main. Instead I am placing this line in the pip.txt
And there are times when the invoke img-buld does not work, as if it were taking from a cache, however if I move some things in the pip.txt it does start to compile, although without any error from what I can see . I see the error later when I start odoo with docker compose up and the library is not found. One of the questions would be if there is a way to force the compilation to take place, even if no changes have been made to pip.txt

**Reply by pleirb** [reply]
The other question would be, what is the possibility that the library is being installed in another python environment that odoo is not seeing?

**Reply by pleirb** [reply, marked as answer]
I think I already achieved it, using the following command:
`docker compose build --no-cache`
It would be nice to enable the --no-cache flag to the invoke img-build command

**Reply by pedrobaeza** [reply]
I would say that any change in the important files will launch the proper building, but we'll check.
